db/db_impl/db_impl.h

```diff
@@ -2575,6 +2576,16 @@ inline Status DBImpl::FailIfTsSizesMismatch(
         << ts_sz << " given";
     return Status::InvalidArgument(oss.str());
   }
+  if (ts_for_read) {
+    auto cfh = static_cast_with_check<ColumnFamilyHandleImpl>(column_family);
+    auto cfd = cfh->cfd();
+    std::string current_ts_low = cfd->GetFullHistoryTsLow();
+    if (!current_ts_low.empty() &&
+        ucmp->CompareTimestamp(ts, current_ts_low) < 0) {
```

Comment:
Nit: use `std::ostringstream` to also print the invalid timestamp and full_history_ts_low to better assist debugging.

Reply:
Good point. Done